hl7apy/mllp.py

```diff
@@ -47,50 +54,54 @@ def __str__(self):
 
 
 class _MLLPRequestHandler(StreamRequestHandler):
+    encoding = 'utf-8'
+
     def __init__(self, *args, **kwargs):
         StreamRequestHandler.__init__(self, *args, **kwargs)
 
     def setup(self):
-        self.sb = "\x0b"
-        self.eb = "\x1c"
-        self.cr = "\x0d"
-        self.validator = re.compile(self.sb + "(([^\r]+\r)*([^\r]+\r?))" + self.eb + self.cr)
+        self.sb = b"\x0b"
+        self.eb = b"\x1c"
+        self.cr = b"\x0d"
+        self.validator = re.compile(
+            self.sb.decode('ascii') + r"(([^\r]+\r)*([^\r]+\r?))" + self.eb.decode('ascii') + self.cr.decode('ascii'))
```

Comment:
Note that this validator can only run against decoded text values (unicode in Python 2.7), so we need to decode the sequence identifiers to build a `str` instead of `bytes`.